### PR TITLE
rofi-top: init at unstable-2017-10-16

### PR DESCRIPTION
**Pull request by aacebedo**
###### Description of changes
Adding the plugin rofi-top.

###### Things done
- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/rofi-top/default.nix [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    ./0001-Patch-plugindir-to-output.patch
+    ./0002-Patch-add-cairo.patch
+  ];
+
+  meta = with lib; {
+    description = "A plugin for rofi that emulates top behaviour";
+    homepage = "https://github.com/davatorium/rofi-top";
+    license = licenses.mit;
+    maintainers = with maintainers; [ aacebedo ];
+    platforms = with platforms; linux;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/rofi-top/default.nix:49]:
```suggestion
    platforms = platforms.linux;
```

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -35947,4 +35947,6 @@ with pkgs;
   swift-corelibs-libdispatch = callPackage ../development/libraries/swift-corelibs-libdispatch { };
 
   swaysettings = callPackage ../applications/misc/swaysettings { };
+
+  rofi-top = callPackage ../applications/misc/rofi-top { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:35951]: Please sort this.

File: pkgs/applications/misc/rofi-top/default.nix [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    glib
+    libgtop
+    rofi-unwrapped
+  ];
+
+  patches = [
+    ./0001-Patch-plugindir-to-output.patch
+    ./0002-Patch-add-cairo.patch
+  ];
+
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/rofi-top/default.nix:43]:
```suggestion
```

File: pkgs/applications/misc/rofi-top/default.nix [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  pname = "rofi-top";
+  version = "unstable-2017-10-16";
+
+  src = fetchFromGitHub {
+    owner = "davatorium";
+    repo = pname;
+    rev = "9416addf91dd1bd25dfd5a8c5f1c7297c444408e";
+    sha256 = "sha256-lNsmx1xirepITpUD30vpcs5slAQYQcvDW8FkA2K9JtU=";
+  };
+
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/rofi-top/default.nix:24]:
```suggestion

  patches = [
    ./0001-Patch-plugindir-to-output.patch
    ./0002-Patch-add-cairo.patch
  ];

```

can we upstream this?

Comment (aacebedo) [reply pkgs/applications/misc/rofi-top/default.nix:24]: Sure, I'll try for the cairo patch.

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (aacebedo) [reply pkgs/top-level/all-packages.nix:35951]: Done I put it next to the other rofi plugins

File: pkgs/applications/misc/rofi-top/default.nix (hunk 1 above)
Comment (aacebedo) [reply pkgs/applications/misc/rofi-top/default.nix:49]: done

File: pkgs/applications/misc/rofi-top/default.nix (hunk 4 above)
Comment (aacebedo) [reply pkgs/applications/misc/rofi-top/default.nix:24]: Got an answer for cairo from the author. Removing this dependency is on his todolist however I don't have an ETA so I think we shall keep the patches for the moment.

File: pkgs/applications/misc/rofi-top/0001-Patch-plugindir-to-output.patch [hunk 5]
```diff
@@ -0,0 +1,15 @@
+diff -u a/configure.ac b/configure.ac
+--- a/configure.ac  2022-07-08 20:18:30.951661098 +0200
++++ b/configure.ac  2022-07-08 20:35:46.865713449 +0200
+@@ -53,8 +53,9 @@
+ dnl ---------------------------------------------------------------------
+ PKG_CHECK_MODULES([glib],     [glib-2.0 >= 2.40 gio-unix-2.0 gmodule-2.0 libgtop-2.0])
+ PKG_CHECK_MODULES([rofi],     [rofi])
++PKG_CHECK_MODULES([cairo],    [cairo cairo-xcb])
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/rofi-top/0001-Patch-plugindir-to-output.patch:8]: This line belongs into the cairo patch.

Comment (aacebedo) [reply pkgs/applications/misc/rofi-top/0001-Patch-plugindir-to-output.patch]: Good point. Changed it.

File: pkgs/top-level/all-packages.nix [hunk 6]
```diff
@@ -35947,4 +35949,5 @@ with pkgs;
   swift-corelibs-libdispatch = callPackage ../development/libraries/swift-corelibs-libdispatch { };
 
   swaysettings = callPackage ../applications/misc/swaysettings { };
+
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:35952]:
```suggestion
```

**Review by SuperSandro2000 — COMMENTED** [review]
Please apply the remaining suggestions

File: pkgs/top-level/all-packages.nix (hunk 6 above)
Comment (aacebedo) [reply pkgs/top-level/all-packages.nix]: Line removed.

File: pkgs/applications/misc/rofi-top/default.nix (hunk 3 above)
Comment (aacebedo) [reply pkgs/applications/misc/rofi-top/default.nix]: patches location moved.

**Pull request merged**